### ISSUE TITLE
classifier predictions use argmax(predict_proba), which discards class_weight (SVM) on imbalanced data

### ISSUE THREAD

**Issue by felixbur** [felixbur/nkululeko#440]
_Found with nkululeko 1.11.0 (datasets 4.8.4, transformers/torch from the venv); code location checked against master 2196be9 (1.11.3)._

**File:** `nkululeko/models/model.py:337-343` (`get_predictions`)

```python
predictions = self.clf.predict_proba(self.feats_test.to_numpy())
...
predictions = probas.idxmax(axis=1).values
```

For `svm` (`SVC(probability=True, class_weight="balanced")`), `predict_proba` comes from a separate Platt-scaling calibration that re-learns the class prior. So `class_weight = True` shifts the SVM decision function, but the argmax over calibrated probabilities throws that shift away, and the model falls back to predicting the majority class.

**Impact (respvox asthma, 16% positive, 1799 test segments, WavLM-large layer-12 embeddings, SVM C=0.1, class_weight=True):**

| Prediction rule | Test UAR | Sensitivity |
|---|---|---|
| nkululeko (argmax of `predict_proba`) | 0.546 | 0.16 |
| same probabilities, threshold at training prior (0.154) | 0.727 | – |
| sklearn `SVC.predict()`, same features/params | 0.750 | 0.80 |

AUC of nkululeko's probabilities is 0.785, i.e. the model ranks well; only the label decision is wrong. The same pattern appears for xls-r (0.584 vs 0.733) and HuBERT (0.522 vs 0.717).

**Suggested fix:** take labels from `self.clf.predict(...)` and keep `predict_proba` only for the probability outputs / uncertainty. Alternatively (or additionally), offer a `[MODEL] threshold` option for binary tasks (e.g. `prior` or tuned on dev).

**Minor, related:** with `[EXP] traindevtest = True`, the final line after the test evaluation still says `best (DEV) result=...`, because `modelrunner.split_name` stays `dev`. The value printed is the test score.

**Comment by felixbur** [comment]
fixed by 1.11.4
